test/irb/test_cmd.rb

```diff
@@ -610,5 +628,57 @@ def test_whereami
       assert_empty err
       assert_match(/^From: .+ @ line \d+ :\n/, out)
     end
+
+    def test_whereami_alias
+      input = TestInputMethod.new([
+        "@\n",
+      ])
+      IRB.init_config(nil)
+      IRB.conf[:COMMAND_ALIASES] = { :'@' => :whereami }
+      workspace = IRB::WorkSpace.new(build_main)
+      irb = IRB::Irb.new(workspace, input)
+      IRB.conf[:MAIN_CONTEXT] = irb.context
+      out, err = capture_output do
+        irb.eval_input
+      end
+      assert_empty err
+      assert_match(/^From: .+ @ line \d+ :\n/, out)
+    end
+
+    def test_vars_with_aliases
+      input = TestInputMethod.new([
+        "@foo\n",
+        "$bar\n",
+      ])
+      IRB.init_config(nil)
+      IRB.conf[:COMMAND_ALIASES] = {
+        :'@' => :whereami,
+        :'$' => :show_source,
+      }
+      main = build_main
+      main.instance_variable_set(:@foo, "foo")
+      $bar = "bar"
+      workspace = IRB::WorkSpace.new(main)
+      irb = IRB::Irb.new(workspace, input)
+      IRB.conf[:MAIN_CONTEXT] = irb.context
+      out, err = capture_output do
+        irb.eval_input
+      end
+      assert_empty err
+      assert_match(/"foo"/, out)
+      assert_match(/"bar"/, out)
+    end
+
+    private
+
+    def build_main
+      main = Object.new
+      main.extend(IRB::ExtendCommandBundle)
+      # Simulate part of the IRB::Irb#run behavior
+      IRB.conf[:COMMAND_ALIASES].each do |alias_name, cmd_name|
```

Comment:
We can get rid of this if we move the command installation to `Irb#initialize`.

Reply:
And if the above 2 comments are correct, we probably won't need this `build_main` anymore.

Reply:
Your patch will stop supporting non-symbol aliases in `IRB.conf[:COMMAND_ALIASES]`.

Reply:
fixed it appropriately in https://github.com/ruby/irb/pull/426/commits/02e9369c059b4c45e829d712bb26610f1a3453b8